airflow/providers/celery/sensors/celery_queue.py

```diff
@@ -74,4 +73,8 @@ def poke(self, context: Context) -> bool:
 
             return reserved == 0 and scheduled == 0 and active == 0
         except KeyError:
-            raise KeyError(f"Could not locate Celery queue {self.celery_queue}")
+            # TODO: remove this if check when min_airflow_version is set to higher than 2.7.1
+            message = f"Could not locate Celery queue {self.celery_queue}"
+            if self.soft_fail:
+                raise AirflowSkipException(message)
+            raise KeyError(message)
```

Comment:
This change fixes the soft_fail only if there is a KeyError exception, but if there is another exception, it will be raised regardless the value of soft_fail.

You need to add something like to make it 100% correct:
```python
        except Exception as e:
            if self.soft_fail:
                raise AirflowSkipException(str(e))
            raise
```

Reply:
Sure. but if we want to keep the current message. we probably need to `raise AirflowExcpetion(message) from e` which will no longer be KeyError. not sure whether it can be an issue

Reply:
I was thinking about this https://github.com/apache/airflow/pull/34474#discussion_r1332038159

Reply:
Sure. Just update it